src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -3299,7 +3299,12 @@ void AuraEffect::HandleModMechanicImmunity(AuraApplication const* aurApp, uint8
     }
 
     if (apply && GetSpellInfo()->HasAttribute(SPELL_ATTR1_DISPEL_AURAS_ON_IMMUNITY))
-        target->RemoveAurasWithMechanic(mechanic, AURA_REMOVE_BY_DEFAULT, GetId());
```

Comment:
Problems with:
 "Blink" - http://wotlk.openwow.com/spell=1953
 "Will of the Forsaken" -http://wotlk.openwow.com/spell=7744
They don't remove CC effects.